### PR TITLE
Beta history annotation, title, and tag editor

### PR DESCRIPTION
**Pull request by Nerdinacan**
There are a lot of minor usability issues with the editing of the title, annotation and tags for a history or a related collection (they share a lot of code). This is going to address the inability to erase annotations, the click-in/click-out clumsiness of the title editor and the tag editor which badly needs a rework.

I think instead of the 3 different kinds of editors all having their own editing modes I might just switch the whole panel from edit to display with a single toggle, let the user change tags, the annotation and the title at the same time then click a button to save it all.

Builds on: https://github.com/galaxyproject/galaxy/pull/12286
https://github.com/galaxyproject/galaxy/pull/12334

#### Desired Api for these components

```html

<!-- while viewing a history -->
<HistoryDetails :history.sync="someHistory" />

<!-- or equivalently -->
<HistoryDetails :history="someHistory" @update:history="someEventHandler" />

<!-- while drilled-down into a collection -->
<Details :dsc="dsc" @update:dsc="updateHandler" :writable="writableFlag"  />

```
These components should be simple editor forms. They should accept a model object (ex: model/History.js) and emit a the update event (to conform with .sync) yielding a fresh History/Collection model object when properties have changed.  Persistence is handled by another component upstream that listens for the update event as it is forwarded up the component tee.

##  Top-level history menus

Removed the priority menu since it was becoming problematic. Now there's simpler dropdowns. Moved slots around to house main history menu to save some of the vertical space that was getting eaten up. The left menu is the new "history list menu" and the right menu is similar to the old gear menu.

gear_menu

history_menu

## Content Filtering

 Cleaned up the UI. Separated the many operations into 3 menus. The "selected" menu is disabled unless the user has created a selection. The "History" menu holds all the bulk assignment operations. The plus on the side is for creating new content like dataset collections using the existing collection assembly modal boxes.

filters_selection

filters_history

filters_new

### Editing

The editing forms were a little problematic. There were strange delays and some legitimate bugs in the update cycle which were failing to get changes into the local cache. Most of these problems stemmed from the fact that previously individual history/dataset fields were being edited a-la cart.

There's now a legitimate temp object for editing that gets manipulated in the form instead of playing games with the caching layer. Updates are now atomic in the sense that the form will take a history in.... the user edits it, and what comes out of the form is a fresh History model instance ready to be cached/saved by the providers. Reliance on Vue's "convenient" mutable observable system resulted in some hard to find bugs, so I've simply adopted a traditional immutable model strategy that does not rely on Vue's watching system for individual object properties.

Changes to a model should now be accomplished by creating a new model object, manipulating it, then emitting it, often like this:

```js
methods: {
    updateDataset(newDatasetProps= {}) {
        const newDs = this.tempData.patch(newDatasetProps); //  patch makes a new immutable dataset instance
        this.$emit('update:dataset', newDs); // we emit the whole new object
    }
}
```

There's now an "edit mode" for all of the details cards which switches the title/annotation/tags into a temporary editing state which can be reverted and cancelled using the menu itmes on the right. The edit menu is on the right hand side of the display area.

editor

filters

### Selenium updates

Most of the selenium failures revolve around the automated "load up a fake history and check that it's there" functions which don't work with the new UI. I'll be updating those here as they relate to the new code as well.

#### Addresses issues:
https://github.com/galaxyproject/galaxy/issues/10947#issuecomment-873103137

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: client/src/components/plugins/icons.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
+import { library } from "@fortawesome/fontawesome-svg-core";
```
Comment (jmchilton) [inline client/src/components/plugins/icons.js:1]: 😍😍😍 . This would save me so much copy-pasted code.

Comment (Nerdinacan) [reply client/src/components/plugins/icons.js:1]: I'm glad you like it. But it shouldn't even be in the javascript at all. Adding a SVG background is well within the capabilities of the css layer.

**Comment by jmchilton** [comment]
This looks really nice - in addition to being very pretty these little widgets - annotation and renaming histories - was where I kept hitting roadblocks in terms of updating the Selenium tests. If this solves those usability issues and we can bring this out of WIP - I'd be happy to pick back up working through the Selenium tests.

**Comment by Nerdinacan** [comment]
@jmchilton 

Sure, I'll convert it. My advice is to add data-XYZ attributes to the markup as identifiers for selenium instead of relying/adding ID and classes which can confound the javascript (when duplicate IDs appear) and the css (when new classes appear that already have definitions). 

Data attributes are a standard part of html and they don't impact the css at all.

I think this strategy is going to have a big payoff if you run with it:

navigation.yml
```
item:
  selectors:
    by_state_and_hid: '.beta.history .listing [data-hid="${hid}"][data-state="${state}"]'
```

**Comment by jmchilton** [comment]
in re: data- attributes - I agree, I converted the state stuff to be encoded that way.

https://github.com/galaxyproject/galaxy/pull/10965/files#diff-aec06b14127dfe1811c74cabf561deaeed45439530493eba8453b8bbe59c6c45R78

And new tests that I've written I've gotten a bit more bold with these:

https://github.com/galaxyproject/galaxy/pull/11160/files#diff-2865407289486eea263e44f295d44a711e9d27e0f7c8d065b3da7f40ff2f85feR400
https://github.com/galaxyproject/galaxy/pull/11160/files#diff-2865407289486eea263e44f295d44a711e9d27e0f7c8d065b3da7f40ff2f85feR403

I think in the past I was a bit too hesitant to make changes to the DOM of other people's components to make it easier to parse for tours/selenium/etc..

**Comment by dannon** [comment]
I reran the jest test that failed, passed fine this time and works reliably locally.  Must have been a blip, but we can keep an eye out for it.

This looks great and is working nicely tinkering with editing stuff locally here -- thanks for all the improvements here.

**Review by dannon — APPROVED** [review]

**Pull request merged**